### DISCUSSION TITLE
[교재] 스터디 코딩 문제 1번_1주차

### DISCUSSION THREAD

**Discussion by kth232** [soeunl/502-3-Study#21] [General]
Q1. 반복문을 사용하여 다음 모양을 출력하는 프로그램을 만들어보세요.(교재 4장 123p 연습문제 Q4)
```
// 출력 결과
   *
  ***
 *****
*******
```

**Comment by heyejiyang** [comment]
```
public static void main(String[] args) {
        int n = 1;
        for(int i=0; i<4;i++){
            for(int k = 0; k<3-i; k++){
                System.out.print(" ");
            }
            for(int j = n; j>0;j--){
                System.out.print("*");
            }
            n+=2;
            System.out.println();
        }

    }
```

**Comment by soeunl** [comment]
![KakaoTalk_20240506_163533221_04](https://github.com/soeunl/502-3-Study/assets/154653812/ae4c0f41-25ba-453b-be02-ecd236db445b)

전 다 왼쪽으로 붙었어요ㅜㅜ

**Comment by siwon9** [comment]
```
package QuestionW1;


import java.util.stream.IntStream;

public class Ex03 {
    public static void main(String[] args) {
        int total = 4;
        IntStream.range(0, total).forEach(i -> {
            System.out.print(" ".repeat(total - i - 1));
            System.out.println("*".repeat(i*2 + 1)); // 줄개행
        });
    }
}

```

**Comment by soeunl** [comment]
public class Star {

	public static void main(String[] args) {

		int n = 5;
		
        for (int i = 1; i <= n; i++) {
        	
        	for (int k = 4; k >= i; k--) {
        		System.out.print(" ");
        	}
            for (int j = 1; j <= i; j++) {
                System.out.print("*");
            }
            for (int j = 2; j <= i; j++) {
            	System.out.print("*");
            }
            System.out.println();
        }
    }
}

for문을 동시에 4개나 돌려서 트리 만들었어요ㅎ

![스크린샷 2024-05-08 224450](https://github.com/soeunl/502-3-Study/assets/154653812/278009cd-1309-4b30-86bb-840a18b763ee)

**Comment by soeunl** [comment]
public class Star {

	public static void main(String[] args) {

		int n = 5;
		
        for (int i = 1; i <= n; i++) {
        	
        	for (int k = 4; k >= i; k--) { 
        		System.out.print(" "); // 공백
        	}
            for (int j = 1; j <= i * 2 - 1; j++) {
                System.out.print("*"); // 별
            }
            
            System.out.println();
        }
    }
}
![스크린샷 2024-05-08 231253](https://github.com/soeunl/502-3-Study/assets/154653812/fe2a9151-1cab-4951-ae52-5a129e2c646b)

for문 하나 줄여서 3개로 만들기

**Comment by kth232** [comment]
```
public class Q1 {
    public static void main(String[] args) {
        int line=4;
        for(int i = 1; i<=line; i++) {
            for(int k = line ; k>=i; k--){
                System.out.print(" ");
            }
            for(int j=1; j <=i*2-1;j++){
                System.out.print("*");
            }
            System.out.println();
        }
    }
}
```
